formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonEncoder.kt

```diff
@@ -130,6 +148,16 @@ internal class StreamingJsonEncoder(
         return true
     }
 
+    override fun encodeInline(inlineDescriptor: SerialDescriptor): Encoder {
+        return if (inlineDescriptor.isUnsignedNumber) StreamingJsonEncoder(
+            ComposerForUnsignedNumbers(
+                composer.sb,
+                composer.json
+            ), json, mode, null
```

Comment:
Why not just passing `modeReuseCache` to keep the code as simple as it has been?

Reply:
Because of https://github.com/Kotlin/kotlinx.serialization/blob/dev/formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonEncoder.kt#L36 and `!==` in it:

Since we re-create `StreamingJsonEncoder` with different composer, it fails reference equality check and in result, cache is populated with encoder+ComposerForUnsigned numbers, which is incorrect

Reply:
Ok, thanks!